src/python/pants/init/plugin_resolver.py

```diff
@@ -21,20 +23,19 @@
 from pants.init.bootstrap_scheduler import BootstrapScheduler
 from pants.option.global_options import GlobalOptions
 from pants.option.options_bootstrapper import OptionsBootstrapper
-from pants.option.subsystem import Subsystem
 from pants.util.logging import LogLevel
 
 logger = logging.getLogger(__name__)
 
 
-S = TypeVar("S", bound=Subsystem)
-
-
 @dataclass(frozen=True)
 class PluginsRequest:
     # Interpreter constraints to resolve for, or None to resolve for the interpreter that Pants is
     # running under.
-    interpreter_constraints: Optional[InterpreterConstraints] = None
+    interpreter_constraints: InterpreterConstraints | None
+    # Requirement constraints to resolve with. If plugins will be loaded into the global working_set
+    # (i.e., onto the `sys.path`), then these should be the current contents of the working_set.
+    constraints: tuple[Requirement, ...]
```

Comment:
Should this be our `PipRequirement`?

Reply:
Maybe...? But it doesn't look like that class does anything right now. cc @benjyw

Reply:
The idea is to allow you to have optional args like `--hash` be preserved.

Reply:
Sure... but it doesn't do that right now.